게시물 상세 API 작성

simseulnyang opened the pull request:
## PR 체크리스트
아래 항목을 확인해 주세요:

- [x] 커밋 메시지가 우리의 가이드라인을 따르고 있는지 확인하세요
- [x] 변경 사항에 대한 테스트가 추가되었는지 확인하세요 (버그 수정 / 기능 추가)
- [ ] 문서가 추가되거나 업데이트되었는지 확인하세요 (버그 수정 / 기능 추가)

## PR 유형
이 PR은 어떤 종류의 변경을 가져오나요?

<!-- 이 PR에 해당하는 것을 "x"로 표시하세요. -->

- [ ] 버그 수정
- [x] 새로운 기능 추가
- [ ] 코드 스타일 업데이트 (서식, 로컬 변수)
- [ ] 리팩터링 (기능 변경 없음, API 변경 없음)
- [ ] 빌드 관련 변경
- [ ] CI 관련 변경
- [ ] 문서 내용 변경
- [ ] 애플리케이션 / 인프라 변경
- [ ] 기타... 설명:

## 현재 동작은 무엇인가요?
<!-- 수정 중인 현재 동작을 설명하거나 관련된 이슈에 링크하세요. -->

이슈 번호: #10

## 새로운 동작은 무엇인가요?
게시물 상세 조회 API 구축
## 이 PR은 호환성 변경을 도입하나요?

- [ ] 예
- [x] 아니요

<!-- 이 PR에 호환성 변경이 포함되어 있다면, 기존 응용 프로그램에 대한 영향과 마이그레이션 경로를 아래에 설명해 주세요. -->

## 기타 정보
![image](https://github.com/Django-Wanted-Internship-3-Team/repo1_feed-service/assets/123173170/2fb9ae08-2564-4bbc-97d7-c37ea66ff0b4)

simseulnyang committed:
:white_check_mark:Test : 게시물 상세 조회 API Test코드 작성 #10

- 해당 content_id가 존재하는 게시물 상세 조회 시 성공 케이스
- 해당 content_id가 존재하지 않는 게시물 상세 조회 시 실패 케이스

simseulnyang committed:
:white_check_mark:Test : 게시물 상세 조회 API Test코드 작성

- 해당 content_id가 존재하는 게시물 상세 조회 시 성공 케이스
- 해당 content_id가 존재하지 않는 게시물 상세 조회 시 실패 케이스

simseulnyang committed:
:sparkles:Feat : 게시물 상세 API 작성 #10

- serializer작성(PostDetailQuerySerializer & PostDetailSerializer)
- 게시글 상세조회 관련 view 생성
- /api/posts/<content_id>에 대한 view 인터페이스 추가

simseulnyang committed:
:sparkles:Feat : 게시물 상세 API 작성 #10

- serializer작성(PostDetailQuerySerializer & PostDetailSerializer)
- 게시글 상세조회 관련 view 생성
- /api/posts/<content_id>에 대한 view 인터페이스 추가

simseulnyang committed:
Merge branch 'feature/issue-010' of https://github.com/Django-Wanted-Internship-3-Team/repo1_feed-service into feature/issue-010

Chestnut90 reviewed (approved):
고생하셨습니다! 😄

테스트 commit 관련해서 현재 test -> refact -> test 순으로 작성되어 있어 보입니다.
만약, 초기 테스트 commit으로 체크아웃시 테스트 코드에는 api에 대한 코드가 작성 되어 있는데 실제 api 코드는 없어서 에러가 발생될 거라고 생각됩니다.

TDD를 적용하고 TDD를 commit에 잘 반영하는 방법을 같이 생각해 보면 좋을 것 같습니다!
저도 관련된 내용을 찾아보곤 있는데 commit 관련한 부분은 정보를 얻기 어렵네요.😭

lfoyh6591 commented:
LGTM 👍

saJaeHyukc commented on posts/views.py:
```diff
@@ -210,3 +213,26 @@ def get_ordering(self, posts: QuerySet[Post], ordering: str) -> QuerySet[Post]:
+    def get(self, request: Request, content_id: str) -> Response:
+        post = get_object_or_404(Post, content_id=content_id)
+
+        # 게시글 조회수 증가
+        post.view_count += 1
+        post.save()
```
해당 부분을 Get 요청을 할 시 Django에서 모듈 중 하나인 signal을 한번 사용해보시는 것은 어떠신가요??

saJaeHyukc replied:
아래는 singal 예제입니다1

https://dgkim5360.tistory.com/entry/django-signal-example

saJaeHyukc commented on posts/views.py:
```diff
@@ -210,3 +213,26 @@ def get_ordering(self, posts: QuerySet[Post], ordering: str) -> QuerySet[Post]:
+
+    @swagger_auto_schema(
+        operation_summary="content_id에 해당하는 게시글 상세 조회",
+        query_serializer=PostDetailQuerySerializer,
```
PostDetailQuerySerialzier같은 경우 제거하셔도 될 것 같습니다. urls.py에서 작성되어있는 query같은 경우에는 drf_yasg에서 자동할당이 되어서 그렇습니다 :)

simseulnyang replied:
아하! 알겠습니다.

saJaeHyukc commented on posts/tests/views/test_posts_detail_view.py:
```diff
@@ -0,0 +1,60 @@
+            path=reverse("post-detail", kwargs={"content_id": self.content_ids[0]}),
+            data={"type": "all", "hashtag": self.hashtag.id, "ordering": "created_at", "search": ""},
+        )
+        print(response.data)  # 서버의 응답 내용 출력
```
사용하지 않는 print 문은 제거해주시면 좋을 것 같습니다 :)

simseulnyang replied:
넵! 바로 지워두겠습니다.

saJaeHyukc replied:
감사합니다 :)

simseulnyang committed:
:recycle:Refactor : 사용하지 않는 일부 코드 삭제

- test_posts_detail_view의 print문 삭제
- PostDetail의 qeury_serializer 삭제
- serializer에 작성한 PostDetailQuerySerializer 삭제

simseulnyang committed:
:sparkles:Feat : Django signal 통해 이벤트 발생 시 조회 수 증가 기능
구현

- signals.py 생성 및 조회수 증가 로직 구현
- apps.py 업데이트하여 signal 활성화
- views.py PostDetail 클래스에 작성한 조회수 증가 코드 삭제
- 조회수 증가와 관련한 testcode 작성

simseulnyang committed:
:sparkles:Feat : Django signal 통해 이벤트 발생 시 조회 수 증가 기능
구현 #10

- signals.py 생성 및 조회수 증가 로직 구현
- apps.py 업데이트하여 signal 활성화
- views.py PostDetail 클래스에 작성한 조회수 증가 코드 삭제
- 조회수 증가와 관련한 testcode 작성

simseulnyang committed:
Merge branch 'feature/issue-010' of https://github.com/Django-Wanted-Internship-3-Team/repo1_feed-service into feature/issue-010

simseulnyang committed:
:recycle:Refactor : signals 모듈을 가져오는 import문 추가

- signals 모듈을 가져오는 import 문을 추가했습니다.
- 이 import 문은 signals이 제대로 로드되고 실행되도록 하는 데 필요합니다.

simseulnyang committed:
:hammer:Chore : testcode 클래스명 변경(PostListViewTest ->
PostDetailViewTest) #10

- 게시물 상세 조회 관련 테스트 코드를 위한 클래스명이어서 관련 명으로
  교체
